api/v1alpha1/accesslogging_types.go

```diff
@@ -92,6 +100,41 @@ type FileEnvoyProxyAccessLog struct {
 	Path string `json:"path,omitempty"`
 }
 
+// ALSEnvoyProxyAccessLog defines the gRPC Access Log Service (ALS) sink.
+// The service must implement the Envoy gRPC Access Log Service streaming API:
+// https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/accesslog/v3/als.proto
+// Access log format information is passed in the form of gRPC metadata when the
+// stream is established. Specifically, the following metadata is passed:
+//
+// - `x-accesslog-text` - The access log format string when a Text format is used.
+//
+// - `x-accesslog-attr` - JSON encoded key/value pairs when a JSON format is used.
+type ALSEnvoyProxyAccessLog struct {
+	// LogName defines the friendly name of the access log to be returned in
+	// StreamAccessLogsMessage.Identifier. This allows the access log server
+	// to differentiate between different access logs coming from the same Envoy.
+	// +optional
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:default="accesslog"
+	LogName string `json:"logName,omitempty"`
+	// Host define the extension service hostname.
+	Host string `json:"host,omitempty"`
+	// Port defines the port the extension service is exposed on.
+	//
+	// +optional
+	// +kubebuilder:validation:Minimum=0
+	Port int32 `json:"port,omitempty"`
```

Comment:
xref https://github.com/envoyproxy/gateway/issues/3041, we should not use host and port in new API.

can you proposal API first, otherwise there will be a lot of waste?

Reply:
api proposal: #3078